### PR TITLE
Add notifications related to currently displayed scenario

### PR DESCRIPTION
**Pull request by mgoworko**
## Describe your changes

The notifications improvements, extracted to separate PR from https://github.com/TouK/nussknacker/pull/7165:
- related to periodic processes, but other processing types too
- before: user did not see any new scenario activities, which were triggered by other user
- now: Nu FE receives all notifications as before, plus notifications related to all activities of currently displayed scenario

## Checklist before merge
- [ ] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [ ] Code is cleaned from temporary changes and commented out lines
- [ ] Parts of the code that are not easy to understand are documented in the code
- [ ] Changes are covered by automated tests
- [ ] Showcase in dev-application.conf added to demonstrate the feature
- [ ] Documentation added or updated
- [ ] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [ ] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

## Release Notes

- **New Features**
	- Enhanced notifications endpoint to support filtering by scenario name.
	- Introduced a new service for fetching scenario activities with improved error handling.
	- Added a new notification type for scenario state updates.

- **Improvements**
	- Streamlined notification handling by removing unnecessary data types.
	- Updated API specifications to reflect changes in notification structures.

- **Bug Fixes**
	- Resolved issues with notification refresh scenarios and improved overall reliability.

- **Tests**
	- Expanded test coverage for notifications and scenario activities, ensuring better validation of functionality.

### CODE REVIEW

**Comment by mgoworko** [comment]
@coderabbitai review

**Review by gskrobisz — APPROVED** [review]
LGTM

File: designer/client/src/http/HttpService.ts [hunk 1]
```diff
@@ -180,8 +180,9 @@ class HttpService {
         this.#notificationActions = na;
     }
 
-    loadBackendNotifications(): Promise<BackendNotification[]> {
-        return api.get<BackendNotification[]>("/notifications").then((d) => {
+    loadBackendNotifications(scenarioName: string | undefined): Promise<BackendNotification[]> {
```
Comment (Dzuming) [inline designer/client/src/http/HttpService.ts:183]: What happens when I'm on the scenarios or components list and there is no scenarioName?

**Review by Dzuming — APPROVED** [review]

File: engine/flink/management/periodic/src/main/scala/pl/touk/nussknacker/engine/management/periodic/PeriodicProcessService.scala [hunk 2]
```diff
@@ -75,7 +76,7 @@ class PeriodicProcessService(
     activities = deploymentsWithStatuses.map { case (deployment, metadata) =>
       ScenarioActivity.PerformedScheduledExecution(
         scenarioId = ScenarioId(processIdWithName.id.value),
-        scenarioActivityId = ScenarioActivityId.random,
+        scenarioActivityId = ScenarioActivityId(longUUID(deployment.id.value)),
```
[the remark this answers is not in the document]
Comment (mgoworko) [reply engine/flink/management/periodic/src/main/scala/pl/touk/nussknacker/engine/management/periodic/PeriodicProcessService.scala]: Done + added comment

File: designer/server/src/test/scala/pl/touk/nussknacker/ui/notifications/NotificationServiceTest.scala [hunk 3]
```diff
@@ -86,7 +100,7 @@ class NotificationServiceTest
     val (deploymentService, notificationService) = createServices(deploymentManager)
 
     def notificationsFor(user: LoggedUser): List[Notification] =
-      notificationService.notifications(user, global).futureValue
+      notificationService.notifications(None)(user, global).futureValue
```
[the remark this answers is not in the document]
Comment (mgoworko) [reply designer/server/src/test/scala/pl/touk/nussknacker/ui/notifications/NotificationServiceTest.scala]: Option replaced with ADT `NotificationsScope`

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/process/scenarioactivity/ScenarioActivityService.scala [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+import pl.touk.nussknacker.ui.process.scenarioactivity.ScenarioActivityService.ScenarioActivityFetchError.NoScenario
+import pl.touk.nussknacker.ui.security.api.LoggedUser
+
+import java.time.Instant
+import scala.concurrent.{ExecutionContext, Future}
+import scala.math.Ordered.orderingToOrdered
+
+class ScenarioActivityService(
+    deploymentManagerDispatcher: DeploymentManagerDispatcher,
+    scenarioActivityRepository: ScenarioActivityRepository,
```
[the remark this answers is not in the document]
Comment (mgoworko) [reply designer/server/src/main/scala/pl/touk/nussknacker/ui/process/scenarioactivity/ScenarioActivityService.scala]: Name changed to `FetchScenarioActivityService` (method inside it was named `def fetchActivities`). Added TODO with description.

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/api/ScenarioActivityApiHttpService.scala [hunk 5]
```diff
@@ -39,7 +39,7 @@ import scala.concurrent.{ExecutionContext, Future}
 
 class ScenarioActivityApiHttpService(
     authManager: AuthManager,
-    scenarioActivityService: ScenarioActivityService,
+    scenarioActivityService: FetchScenarioActivityService,
```
Comment (arkadius) [inline designer/server/src/main/scala/pl/touk/nussknacker/ui/api/ScenarioActivityApiHttpService.scala:42]: let's rename parameter as well

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/notifications/NotificationService.scala [hunk 6]
```diff
@@ -1,45 +1,67 @@
...
+    scenarioActivityService: ScenarioActivityService,
     scenarioActionRepository: ScenarioActionRepository,
     dbioRunner: DBIOActionRunner,
     config: NotificationConfig,
     clock: Clock = Clock.systemUTC()
 ) extends NotificationService {
 
-  override def notifications(implicit user: LoggedUser, ec: ExecutionContext): Future[List[Notification]] = {
+  override def notifications(
+      processNameOpt: Option[ProcessName]
```
[the remark this answers is not in the document]
Comment (arkadius) [reply designer/server/src/main/scala/pl/touk/nussknacker/ui/notifications/NotificationService.scala]: Just to be sure `NotificationsForLoggedUser` contains `NotificationsForLoggedUserAndScenario` for every scenario?

**Review by arkadius — APPROVED** [review]
LGTM, merge after fixing one comment and replying comment

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/api/ScenarioActivityApiHttpService.scala (hunk 5 above)
Comment (mgoworko) [reply designer/server/src/main/scala/pl/touk/nussknacker/ui/api/ScenarioActivityApiHttpService.scala]: done

**Pull request merged**